### PR TITLE
Prefer OutputRid over PackageRID for output artifacts.

### PR DESCRIPTION
**Pull request by tmds**
This is a non-functional change that makes it more visible these are output artifacts.

@ViktorHofer @am11 ptal.

### CODE REVIEW

**Comment by tmds** [comment]
I was triggered to make this change because I'd rather see `microsoft.netcore.app.runtime.fedora.36-x64` under `artifacts/bin` for a non-portable build than `microsoft.netcore.app.runtime.linux-x64`.

> @ViktorHofer @am11 ptal.

CI should take a look first to verify I didn't break anything.

File: src/libraries/System.Net.Http/tests/StressTests/HttpStress/Directory.Build.props [hunk 1]
```diff
@@ -12,6 +12,6 @@
         <NetCoreAppCurrentVersion>7.0</NetCoreAppCurrentVersion>
         <MicrosoftNetCoreAppFrameworkName>Microsoft.NETCore.App</MicrosoftNetCoreAppFrameworkName>
         <MicrosoftNetCoreAppRefPackDir Condition="'$(MicrosoftNetCoreAppRefPackDir)' == ''" >$(RepositoryRoot)artifacts/bin/microsoft.netcore.app.ref/</MicrosoftNetCoreAppRefPackDir>
-        <MicrosoftNetCoreAppRuntimePackDir Condition="'$(MicrosoftNetCoreAppRuntimePackDir)' == ''">$(RepositoryRoot)artifacts/bin/microsoft.netcore.app.runtime.$(PackageRid)/$(Configuration)/</MicrosoftNetCoreAppRuntimePackDir>
+        <MicrosoftNetCoreAppRuntimePackDir Condition="'$(MicrosoftNetCoreAppRuntimePackDir)' == ''">$(RepositoryRoot)artifacts/bin/microsoft.netcore.app.runtime.$(OutputRid)/$(Configuration)/</MicrosoftNetCoreAppRuntimePackDir>
```
Comment (ViktorHofer) [inline src/libraries/System.Net.Http/tests/StressTests/HttpStress/Directory.Build.props:15]: This is a stand-alone project which duplicates all the logic. Does it have access to the OutputRid property?

File: src/libraries/System.Net.Security/tests/StressTests/SslStress/Directory.Build.props [hunk 2]
```diff
@@ -12,6 +12,6 @@
         <NetCoreAppCurrentVersion>7.0</NetCoreAppCurrentVersion>
         <MicrosoftNetCoreAppFrameworkName>Microsoft.NETCore.App</MicrosoftNetCoreAppFrameworkName>
         <MicrosoftNetCoreAppRefPackDir Condition="'$(MicrosoftNetCoreAppRefPackDir)' == ''" >$(RepositoryRoot)artifacts/bin/microsoft.netcore.app.ref/</MicrosoftNetCoreAppRefPackDir>
-        <MicrosoftNetCoreAppRuntimePackDir Condition="'$(MicrosoftNetCoreAppRuntimePackDir)' == ''">$(RepositoryRoot)artifacts/bin/microsoft.netcore.app.runtime.$(PackageRid)/$(Configuration)/</MicrosoftNetCoreAppRuntimePackDir>
+        <MicrosoftNetCoreAppRuntimePackDir Condition="'$(MicrosoftNetCoreAppRuntimePackDir)' == ''">$(RepositoryRoot)artifacts/bin/microsoft.netcore.app.runtime.$(OutputRid)/$(Configuration)/</MicrosoftNetCoreAppRuntimePackDir>
```
Comment (ViktorHofer) [inline src/libraries/System.Net.Security/tests/StressTests/SslStress/Directory.Build.props:15]: same here

File: src/libraries/pretest.proj [hunk 3]
```diff
@@ -63,10 +63,10 @@
           Outputs="$(MicrosoftNetCoreAppRuntimePackDir)data\PlatformManifest.txt"
           Condition="'$(BuildTargetFramework)' == '$(NetCoreAppCurrent)' or '$(BuildTargetFramework)' == ''">
     <GenerateFileVersionProps Files="@(SharedFrameworkRuntimeFile)"
-                              PackageId="$(MicrosoftNetCoreAppFrameworkName).Runtime.$(PackageRID)"
+                              PackageId="$(MicrosoftNetCoreAppFrameworkName).Runtime.$(OutputRid)"
```
Comment (ViktorHofer) [inline src/libraries/pretest.proj:66]: This changes the non-shipping "fake" runtime pack id. Do we also make changes to the actual shipping runtime pack sfxproj? I'm suprised that there aren't any changes to the sfxprojs.

Comment (tmds) [reply src/libraries/pretest.proj:66]: I wasn't aware there's a fake runtime pack id, and a shipping runtime pack id.
Do these represent different builds?

I assume the sfxprojs pick up the `MicrosoftNetCoreAppRuntimePack*` properties?

**Comment by ViktorHofer** [comment]
We need to retrigger CI as the previous build isn't available anymore.

**Comment by tmds** [comment]
The unsuccessful jobs failed in the 'Send to Helix' step. I don't see any build or test failures related to the PR changes.

**Comment by ViktorHofer** [comment]
Sorry that it took me longer to reply. Looking into this comment right now:

> I'm not sure what I should validate for https://github.com/dotnet/runtime/pull/76871#discussion_r992403036. Is green CI enough?
Can we trigger CI also for the unknown Banana rid? Azure shows it greyed out, so I think it doesn't run by default.

Need to figure out where the rid is and how to trigger it.

**Comment by ViktorHofer** [comment]
I couldn't find a good way to trigger the banana rid as it had a condition on it to only run for rolling builds. I just removed that condition to get CI validation. We can revert my commit before merging.

**Comment by tmds** [comment]
The banana rid build passed: https://dev.azure.com/dnceng-public/public/_build/results?buildId=80118&view=logs&j=98fdc254-13b3-5066-1dc8-532bea57cf83. Other CI jobs look good too.

I removed the commit that triggered the banana rid job.

This should be good to merge.

**Review by ViktorHofer — APPROVED** [review]

**Comment by ViktorHofer** [comment]
Thanks @tmds

**Pull request merged**